Add task_log_buffer_size option to logger documentation (#4899)
OttoWinter/esphomedocs: components/logger.rst

-  **tx_buffer_size** (*Optional*, int): The size of the buffer used
   for log messages. Decrease this if you’re having memory problems.
   Defaults to ``512``.

> -  **buffer_size** (*Optional*, int): **ESP32 only**: The size of the internal thread-safe ring buffer for log messages.

Author: ```suggestion
-  **task_log_buffer_size** (*Optional*, int): **ESP32 only**: The size of the internal thread-safe ring buffer for task log messages.
```

>    This helps prevent API disconnections when multiple threads attempt to log simultaneously.

Author: ```suggestion
   This prevents API disconnections when multiple threads attempt to log simultaneously.
```

>    Set to ``0`` to disable the log buffer. Defaults to ``576``.

Author: ```suggestion
   Set to ``0`` to disable the log buffer. Defaults to ``768B``.
```
